drivers/ipm/ipm_psoc6.c

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (c) 2018, Cypress Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <device.h>
+#include <string.h>
+#include <ipm.h>
+#include <soc.h>
+#include "cy_sysint.h"
+#include "cy_ipc_drv.h"
+#include "ipm_psoc6.h"
+
+typedef enum {
+	PSOC6_IPM_SERVER = 0,
+	PSOC6_IPM_CLIENT = 1
+} psoc6_ipm_instance_role_t;
+
+struct psoc6_ipm_config_t {
+	u32_t channel;
+	s32_t int_base;
+	s32_t notify_chn;
+	s32_t release_chn;
+	psoc6_ipm_instance_role_t role;
+};
+
+struct psoc6_ipm_data {
+	ipm_callback_t callback;
+	void *callback_ctx;
+};
+
+static void psoc6_ipm_notify_isr(void *arg)
+{
+	struct device *dev = arg;
+	struct psoc6_ipm_data *data = dev->driver_data;
+	const struct psoc6_ipm_config_t *config = dev->config->config_info;
+	u32_t value;
+	u32_t interruptMasked;
```

Comment:
Another issue, taking into account that @galak -1'ed it for now anyway. camelCase naming is not natural Zephyr naming convention. I'd suggest to stick to standard_naming_convention (otherwise we strange mix-up, like just 5 above lines show).